mllib/src/main/scala/org/apache/spark/mllib/evaluation/BinaryClassificationMetrics.scala

```diff
@@ -165,13 +165,15 @@ class BinaryClassificationMetrics @Since("3.0.0") (
     confusions: RDD[(Double, BinaryConfusionMatrix)]) = {
     // Create a bin for each distinct score value, count weighted positives and
     // negatives within each bin, and then sort by score values in descending order.
-    val counts = scoreLabelsWeight.combineByKey(
+    val binnedWeights = scoreLabelsWeight.combineByKey(
       createCombiner = (labelAndWeight: (Double, Double)) =>
         new BinaryLabelCounter(0.0, 0.0) += (labelAndWeight._1, labelAndWeight._2),
       mergeValue = (c: BinaryLabelCounter, labelAndWeight: (Double, Double)) =>
         c += (labelAndWeight._1, labelAndWeight._2),
       mergeCombiners = (c1: BinaryLabelCounter, c2: BinaryLabelCounter) => c1 += c2
-    ).sortByKey(ascending = false)
+    )
+    binnedWeights.persist()
```

Comment:
`binnedWeights` RDD needs to be persisted. At Line 176: `sortByKey()` and Line 216: `collect()` is applied.

Reply:
This has to be unpersisted though. We also customarily only persist if the user input was persisted, as a sort of signal that the user is OK consuming memory/disk. For small inputs this may not be worth it so maybe the user doesn't want the overhead of persisting.